spec/requests/providers/outstanding_mortgage_spec.rb

```diff
@@ -18,40 +18,90 @@
+            expect(response).to redirect_to providers_legal_aid_application_shared_ownership_path(application)
+          end
+        end
+      end
+
+      context 'when an outstanding mortgage value is not entered' do
         let(:application) { legal_aid_application }
+        let(:params) do
+          {
+            legal_aid_application: { outstanding_mortgage_amount: '' },
+            legal_aid_application_id: application.id
+          }
+        end
+
+        it 'shows an error message' do
+          expect(response.body).to include('govuk-error-summary__title')
+        end
 
-        it 'records the value in the legal aid application table' do
+        it 'does not record the value in the legal aid application table' do
           legal_aid_application.reload
-          expect(legal_aid_application.outstanding_mortgage_amount).to be_within(0.01).of(321_654.87)
-          expect(legal_aid_application.updated_at.utc.to_i).to be_within(1).of(Time.now.to_i)
+          expect(legal_aid_application.outstanding_mortgage_amount).to be nil
         end
       end
     end
 
-    context 'when an outstanding mortgage value is not entered' do
-      let(:application) { legal_aid_application }
-      let(:params) do
-        {
-          legal_aid_application: { outstanding_mortgage_amount: '' },
-          legal_aid_application_id: application.id
-        }
+    context 'Submitted with the Save as draft button' do
+      before do
+        params.delete('continue-button')
+        params['draft-button'] = 'Save as draft'
+        patch providers_legal_aid_application_outstanding_mortgage_path(application), params: params
       end
 
-      it 'shows an error message' do
-        expect(response.body).to include('govuk-error-summary__title')
+      context 'when an outstanding mortgage value is entered' do
+        context 'with valid values' do
+          let(:application) { legal_aid_application }
+
+          it 'records the value in the legal aid application table' do
+            legal_aid_application.reload
+            expect(legal_aid_application.outstanding_mortgage_amount).to be_within(0.01).of(321_654.87)
+            expect(legal_aid_application.updated_at.utc.to_i).to be_within(1).of(Time.now.to_i)
+          end
+
+          it 'displays the provider applications home page' do
+            expect(response).to redirect_to providers_legal_aid_applications_path
+          end
+        end
       end
 
-      it 'does not record the value in the legal aid application table' do
-        legal_aid_application.reload
-        expect(legal_aid_application.outstanding_mortgage_amount).to be nil
+      context 'when an outstanding mortgage value is not entered' do
+        let(:application) { legal_aid_application }
+        let(:params) do
+          {
+            legal_aid_application: { outstanding_mortgage_amount: '' },
+            legal_aid_application_id: application.id
+          }
+        end
```

Comment:
If `outstanding_mortgage_amount` was defined as a variable:
```ruby
let(:outstanding_mortgage_amount) { 321_654.87 }
```
And then params defined as:
```ruby
let(:button_params) { 'continue-button' => 'Continue' }
let(:params) do
      {
        legal_aid_application: { outstanding_mortgage_amount: outstanding_mortgage_amount },
        legal_aid_application_id: application.id
      }.merge(button_params)
    end
```
Then rather than having to redefine the whole params to test variations in `outstanding_mortgage_amount` input, all that's needed is to set a different value here. So:
```ruby
let(:outstanding_mortgage_amount) { '' }
```

Reply:
done